core/dbt/contracts/graph/model_config.py

```diff
@@ -445,6 +464,9 @@ class NodeConfig(NodeAndTestConfig):
     # sometimes getting the Union order wrong, causing serialization failures.
     unique_key: Union[str, List[str], None] = None
     on_schema_change: Optional[str] = "ignore"
+    on_configuration_change: OnConfigurationChangeOption = field(
```

Comment:
Make the validation for `on_configuration_change` match that of `on_schema_change`.

Reply:
`on_schema_change` does not appear to be controlled anywhere besides in jinja. the valid values are enforced in jinja. So this is effectively an Enum. I've removed the meta entry on `on_configuration_change` to match.